blockchain/validators/src/config.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Config for validators.
+
+use crate::Error;
+use mc_blockchain_types::BlockIndex;
+use mc_crypto_keys::Ed25519Public;
+use serde::{Deserialize, Deserializer, Serialize};
+use std::{ffi::OsStr, fs, path::Path};
+
+/// A declaration that a pub key is valid for a specified range of block
+/// indexes.
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+pub struct NodeKeyValidity {
+    /// The pub key, as a PEM file path or 64 hexadecimal characters.
+    #[serde(deserialize_with = "deserialize_pem_or_hex")]
+    pub message_signing_pub_key: Ed25519Public,
+    /// The first block index that used this key.
+    pub first_block_index: BlockIndex,
+    /// The last block index that used this key. Can be `None` for current keys.
+    pub last_block_index: Option<BlockIndex>,
+}
+
+impl NodeKeyValidity {
+    /// Instantiate an instance.
+    pub fn new(
+        message_signing_pub_key: Ed25519Public,
+        first_block_index: BlockIndex,
+        // Allows passing `N`, `Some(N)` or `None`
+        last_block_index: impl Into<Option<BlockIndex>>,
+    ) -> Self {
+        Self {
+            message_signing_pub_key,
+            first_block_index,
+            last_block_index: last_block_index.into(),
+        }
+    }
+
+    /// Check whether a given block index falls within the range of this
+    /// [NodeKeyValidity].
+    pub fn index_in_range(&self, block_index: BlockIndex) -> bool {
+        block_index >= self.first_block_index &&
+        // If there is a last block, enforce that it's no greater than that.
+        self.last_block_index.map(|last_index| block_index <= last_index).unwrap_or(true)
```

Comment:
I think the intermediate comment is confusing cargo fmt, or perhaps my previous suggestion here was malformated
```suggestion
        // If there is a last block, enforce that it's no greater than that.
        block_index >= self.first_block_index
            && self
                .last_block_index
                .map(|last_index| block_index <= last_index)
                .unwrap_or(true)
```